s3/replication/replicator/tests/system/load_transfer_test.py

```diff
@@ -281,50 +282,49 @@ async def run_load_test():
     jobs_running = True  # if at least one job is running, keep polling.
     # Max polling iterations to avoid infinite loop
     polling_count = test_config.polling_count
+
+    response = {}
     while jobs_running and polling_count != 0:
-        job_status_task_list = []
-        for job_id in posted_jobs_set:
-            logger.debug(
-                "Checking status for job: job_id= {}".format(job_id))
-
-            job_status_task = asyncio.ensure_future(replicator_session.get(
-                url + '/jobs/' + job_id))
-            job_status_task_list.append(job_status_task)
-
-        # jobs status contains list of response from GET /jobs/<job_id> for
-        # each
-        logger.debug("Waiting for all GET {}/jobs/<job_id> response...".
-                     format(url))
-        get_job_response_list = await asyncio.gather(*job_status_task_list)
-        # logger.debug("get_job_response_list: {}".format(
-        #     get_job_response_list))
-
-        for get_job_resp in get_job_response_list:
-            job_status = await get_job_resp.json()
-            job_id = job_status["job"]["job_id"]
-            job_state = job_status["job"]["state"]
-            logger.debug(
-                "job_state = {} for job_id {}".format(
-                    job_state, job_id))
-            if job_state != "RUNNING":
-                # remove completed job from polling list
-                posted_jobs_set.remove(job_id)
-
-        if len(posted_jobs_set) == 0:
-            # No jobs pending.
+        async with replicator_session.get(
+                url + '/jobs?count&inprogress') as resp:
+            logger.info(
+                'GET jobs returned http Status: {}'.format(resp.status))
+            response = await resp.json()
+
+        logger.info("jobs is : {}".format(response))
+        inprogress_count = response['inprogress-count']
+        logger.info("inprogress_count is : {}".format(inprogress_count))
+
+        if inprogress_count == 0:
+            # No jobs pending then exit here.
             jobs_running = False
             logger.debug("All jobs completed.")
+            sys.exit(0)
         else:
             # There are atleast some running jobs, give time to complete.
             logger.debug(
                 "Pending status for total {} jobs.".format(
-                    len(posted_jobs_set)))
+                    inprogress_count))
             logger.info("Waiting for {} secs before polling job status...".
                         format(test_config.polling_wait_time))
             time.sleep(test_config.polling_wait_time)
 
         polling_count -= 1
 
+    # Execute this if all jobs are not completed.
+    if inprogress_count != 0:
+        async with replicator_session.get(
```

Comment:
Seems redundant since we earlier already got in-progress count. We can just show the url to access to get in-progress list.

Reply:
done